mavros_extras/src/plugins/mount_control.cpp

```diff
@@ -0,0 +1,87 @@
+/**
+ * @brief Mouny Control plugin
+ * @file mount_control.cpp
+ * @author Jaeyoung Lim <jaeyoung@auterion.com>
+ *
+ * @addtogroup plugin
+ * @{
+ */
+/*
+ * Copyright 2019 Jaeyoung Lim.
+ *
+ * This file is part of the mavros package and subject to the license terms
+ * in the top-level LICENSE file of the mavros repository.
+ * https://github.com/mavlink/mavros/tree/master/LICENSE.md
+ */
+
+#include <mavros/mavros_plugin.h>
+
+#include <mavros_msgs/MountControl.h>
+
+namespace mavros {
+namespace extra_plugins {
+
+//! Mavlink enumerations
+using mavlink::common::MAV_MOUNT_MODE;
+using mavlink::common::MAV_CMD;
+using utils::enum_value;
+
+/**
+ * @brief Mount Control plugin
+ *
+ * Publishes Mission commands to control the camera or antenna mount.
+ * @see status_cb()
+ */
+class MountControlPlugin : public plugin::PluginBase {
+public:
+	MountControlPlugin() : PluginBase(),
+	mount_nh("~mount_control")
+	{ }
+
+	void initialize(UAS &uas_)
+	{
+		PluginBase::initialize(uas_);
+
+		command_sub = mount_nh.subscribe("command", 10, &MountControlPlugin::command_cb, this);
+	}
+
+	Subscriptions get_subscriptions()
+	{
+		return { /* Rx disabled */ };
+	}
+
+private:
+	ros::NodeHandle mount_nh;
+	ros::Subscriber command_sub;
+
+	/**
+	 * @brief Send mount control commands to vehicle
+	 *
+	 * Message specification: https://mavlink.io/en/messages/common.html#MAV_CMD_DO_MOUNT_CONTROL
+	 * @param req	received MountControl msg
+	 */
+	void command_cb(const mavros_msgs::MountControl::ConstPtr &req)
+	{
+		mavlink::common::msg::COMMAND_LONG cmd {};
+
+		const uint8_t tgt_sys_id = m_uas->get_tgt_system();
+		const uint8_t tgt_comp_id = m_uas->get_tgt_component();
+		cmd.target_system = tgt_sys_id;
+		cmd.target_component = tgt_comp_id;
```

Comment:
You don't need intermediate `tgt_*` variables.

Reply:
I have addressed the comments :)